### PR TITLE
GUI

### LINKED ISSUE #24: Create a GUI version of the bot
Implement a local version of the bot that can be started through an interface using Java Swing GUI.

### PR DESCRIPTION
**Pull request by christianjokhaji**
Please review.

I had to make some modifications to Main and RecipeApiHandler to account for different app versions (GUI and non-GUI).

### CODE REVIEW

File: src/main/java/ca/unknown/bot/app/Main.java
Comment (xinyue296) [inline src/main/java/ca/unknown/bot/app/Main.java:1]: Amazing work! I think it's a great idea to add the GUI feature to make our bot also a template bot. However, I wonder whether the GUI window always pops out when the cloud or users(who clone our project)? Currently, our bot on the cloud is running the version on the main branch, but I'm not sure if we can make the bot (on the cloud machine) to click the GUI button.

**Review by xinyue296 — CHANGES_REQUESTED** [review]

File: src/main/java/ca/unknown/bot/app/Main.java
Comment (christianjokhaji) [reply src/main/java/ca/unknown/bot/app/Main.java:1]: This basically boils down to which class is chosen as the main class (i.e., either Main.java or GUI.java) in the run configuration:

Screenshot 2024-08-02 at 1 15 37 PM

I'm assuming there's a way to select this from the cloud provider, but not sure. Let me know if there isn't.

Comment (xinyue296) [reply src/main/java/ca/unknown/bot/app/Main.java:1]: yes, now i understand! The cloud machine is  running Main.java (as specified in the build.gradle.kts) when building with gradlew :) so it should be fine

**Review by xinyue296 — APPROVED** [review]

**Review by Overpoweredplatypus — APPROVED** [review]
Thank you so much for the hard work!

**Review by sailoraria — APPROVED** [review]
Thank you for working on this! I'm excited to use this.

**Pull request merged**